tmt/steps/report/__init__.py

```diff
@@ -1,21 +1,25 @@
+from typing import Any, List, Optional
+
 import click
 
 import tmt
+import tmt.steps
 
 
 class Report(tmt.steps.Step):
     """ Provide test results overview and send reports. """
 
     # Default implementation for report is display
     how = 'display'
+    data: List[tmt.steps.StepData]
 
-    def wake(self):
+    def wake(self) -> None:
         """ Wake up the step (process workdir and command line) """
         super().wake()
 
         # Choose the right plugin and wake it up
         for data in self.data:
-            plugin = ReportPlugin.delegate(self, data)
+            plugin = ReportPlugin.delegate(self, dict(data))
```

Comment:
Because method `ReportPlugin.delegate()` expects `Dict[Any, Any]` which is not compatible with `tmt.steps.StepData` (derived from `TypedDict`):

```
tmt/steps/report/__init__.py:22: error: Argument 2 to "delegate" of "Plugin" has incompatible type "StepData"; expected "Dict[Any, Any]"
```

See e.g. https://github.com/python/mypy/issues/4976.

I agree that converting to `dict` is not so pretty. Perhaps changing the method to `def delegate(..., data: Union[Dict[Any, Any], StepData])` could work.